src/lib/molecules/Charts/MapChartD3/MapChartD3.tsx

```diff
@@ -0,0 +1,697 @@
+                    : null;
+
+            if (region && colorAxis) {
+                const dataClass = legends.find((axis) => region.value >= axis.from && region.value <= axis.to);
+                return {
+                    ...item,
+                    properties: {
+                        ...item.properties,
+                        value: region.value,
+                        color: dataClass?.color,
+                        legend: dataClass?.name
+                    }
+                };
+            } else {
+                return { ...item, properties: { ...item.properties, value: 0, color: defaultFeatureColor } };
+            }
+        });
+
+        setChartData(_chartData);
+    }, [isLoading, withLegend, isViewActive, withActivity, screenType, title, windowWidth, windowHeight, mapData]);
+
+    const handleMouseEnter = (geo: IMapChartFeature, e: MouseEvent): void => {
+        withActivity && setSelectedName(geo.properties.name);
+        const tooltipContent = tooltipRenderer && tooltipRenderer(geo);
+        tooltipRenderer && setTooltipData({ top: e.clientY, left: e.clientX, content: tooltipContent });
+    };
+
+    const handleMouseLeave = () => {
+        withActivity && !isMobile && setSelectedName(null);
+        tooltipRenderer && setTooltipData({ top: 0, left: 0, content: undefined });
+    };
+
+    const handleMouseMove = (event: MouseEvent) => {
+        if (!contextRef.current || !pathRef.current) {
+            return;
+        }
+
+        const context = contextRef.current;
+        const path = pathRef.current;
+
+        const [x, y] = pointer(event);
+
+        // START Find hovered region
+        context.beginPath();
+        const generateContext = path.context(context);
+        const hoveredRegion = chartData.find((feature) => {
+            generateContext(feature as GeoPermissibleObjects);
+
+            return context.isPointInPath(
+                (x - transformRef.current.x) / transformRef.current.k,
+                (y - transformRef.current.y) / transformRef.current.k
+            );
+        });
+
+        // END Find hovered region
+
+        if (!hoveredRegion) {
+            if (!hoveredRegionRef.current) {
+                return;
+            }
+            hoveredRegionRef.current.properties.brightness = 0;
+            drawMap();
+            hoveredRegionRef.current = null;
+            handleMouseLeave();
+            return;
+        }
+
+        // START Highlight regions depends on brightness
+
+        if (hoveredRegion.properties.value && hoveredRegion.properties.color !== defaultFeatureColor) {
+            if (hoveredRegionRef.current?.id === hoveredRegion.id) {
+                handleMouseEnter(hoveredRegion, event);
+
+                if (hoveredRegionRef.current.properties.brightness === hoveredRegion.properties.brightness) return;
+
+                hoveredRegion.properties.brightness = 0;
+                drawMap();
+            } else {
```

Comment:
I think you can remove else

Reply:
We need here `else` variant for `if` because on body of `if` condition we return depending case.